rewiringamerica/embed.rewiringamerica.org
# Break dependency on lit/localize

## Description

This is unfortunately a huge PR, so here's a detailed guide.

Using `@lit/localize` at runtime also entails depending on the
entirety of Lit; the localize package is not factored so that you can
just use the runtime string-substitution part. So we gotta get rid of it.

The idea here was to be minimally disruptive by simply swapping out
the `msg` and `str` functions from lit-localize with our own
implementations.  That's in `src/i18n/use-translated.ts` and
`src/i18n/str.ts`. The `strings:build` command now also invokes sed to
adjust the import in the generated translated-strings files.

Rather than putting the current locale in global state like Lit does,
I put it in React context, accessed via the new custom hook
`useTranslated`.  The hook returns the `msg` function, bound to the
current locale; callsites then use that msg function as before.

Most of the changed files are just plumbing in the new hook and/or msg
function. The most substantive changes are in:

- `state-calculator.tsx` where the locale context is set

- `calculator.tsx` -- this is the old embed, which was never actually
  localized; I removed all the `msg` calls here since we're getting
  rid of it Soon™

- `i18n/use-translated.ts` -- the core of the new stuff

## Test Plan

- Run `yarn strings:extract` and make sure there are no changes to es.xlf.

- Run `yarn strings:build` and make sure there are no changes.

- View the calculator and hit Calculate; make sure all strings come up
  looking correct in English. Add the `lang="es"` attribute to the
  calculator element and make sure the UI immediately switches to
  Spanish, except for incentive blurbs (which come from the API).

  Hit Calculate again and make sure the incentive blurbs are now in Spanish.
